btr-api/src/btr_api/resources/submission.py

```diff
@@ -58,18 +59,36 @@ def registers(id: int | None = None):  # pylint: disable=redefined-builtin
     """Get the submissions, or sepcific submission by id."""
     try:
         if id:
-            if submission := Submission.find_by_id(id):
+            if submission := SubmissionModel.find_by_id(id):
                 return jsonify(type=submission.type, submission=submission.payload)
             return {}, HTTPStatus.NOT_FOUND
 
-        submissions = Submission.get_filtered_submissions()
+        submissions = SubmissionModel.get_filtered_submissions()
+        submissions = [SubmissionSerializer.to_dict(submission) for submission in submissions]
 
         return jsonify(submissions)
 
     except Exception as exception:  # noqa: B902
         return exception_response(exception)
 
 
+@bp.route("/entity/<business_identifier>", methods=("GET",))
+def latest_submission_for_entity(business_identifier: str):
+    """Get the latest submission for specified business identifier."""
+    sf = SubmissionFilter()
+    sf.business_identifier = business_identifier
+
+    try:
+        submissions = SubmissionModel.get_latest_submissions(submission_filter=sf)
+        if submissions:
+            return jsonify(SubmissionSerializer.to_str(submissions[0]))
+
+        return None, HTTPStatus.NOT_FOUND
```

Comment:
I think this throws an uncaught error -- might need to change this to '{}' instead of 'None'